src/sync/reference_counter.zig

```diff
@@ -84,3 +162,13 @@ test "sync.ref_counter: ReferenceCounter works" {
     try std.testing.expect(!x.release());
     try std.testing.expect(x.release());
 }
+
+test "RcSlice payload has the correct data" {
```

Comment:
It would be nice to also have tests covering the reference counting implementation.

Tried writing one as follows but was getting segfault

```
test "RcSlice reference counting" {
    const sig = @import("../sig.zig");
    const slice1 = try RcSlice(u8).alloc(std.testing.allocator, 5);
    defer slice1.deinit(std.testing.allocator);

    // Copy data into the slice1's payload
    @memcpy(slice1.payload(), "hello");

    // Acquire a second reference to the RcSlice
    const slice2 = slice1.acquire();

    // Assert that both slices see the same payload
    const hello: *const [5]u8 = "hello";
    try std.testing.expect(sig.utils.types.eql(hello, slice1.payload()[0..5]));
    try std.testing.expect(sig.utils.types.eql(hello, slice2.payload()[0..5]));

    // Modify the payload via the second reference
    slice2.payload()[0] = 'H';

    // Assert the change is visible through the first reference
    const hello2: *const [5]u8 = "Hello";
    try std.testing.expect(sig.utils.types.eql(hello2, slice1.payload()[0..5]));
    try std.testing.expect(sig.utils.types.eql(hello2, slice2.payload()[0..5]));

    // Release the first reference
    slice1.deinit(std.testing.allocator);

    // The second reference should still be valid
    try std.testing.expect(sig.utils.types.eql(hello2, slice2.payload()[0..5]));
    // Release the final reference
    slice2.deinit(std.testing.allocator);
}
```

Not sure if the issue is in the way I am attempting the test

Reply:
there's a segfault because you deinit it three times. the test passes if you remove the defer near the beginning.

I made another tweak to your test to validate exactly when the pointer is freed.

e8653ff09b526f2a11132e9e9041a58f9cd2f53c